leaktest.go

```diff
@@ -13,13 +13,25 @@ import (
 	"context"
 	"runtime"
 	"sort"
+	"strconv"
 	"strings"
 	"time"
 )
 
+type goroutine struct {
+	id    uint64
+	stack string
+}
+
+type goroutineByID []goroutine
```

Comment:
I'm using the older sort interface because the Travis CI tests also run on Go 1.7